app/src/main/java/com/itsaky/androidide/MainActivity.java

```diff
@@ -118,5 +124,26 @@ protected void onDestroy() {
     binding = null;
   }
 
-  private void showMoveRequired() {}
+  private void showInstallJdkSdkRequired() {
+    if (!Environment.JAVA.exists() | !Environment.ANDROID_HOME.exists()) {
+      final MaterialAlertDialogBuilder builder = DialogUtils.newMaterialDialogBuilder(this);
+      builder.setTitle(R.string.title_warning);
+      TextView view = new TextView(this);
+      view.setPadding(10, 10, 10, 10);
+      view.setText(
+          HtmlCompat.fromHtml(
+              getString(R.string.msg_require_install_jdk_and_android_sdk),
+              HtmlCompat.FROM_HTML_MODE_COMPACT));
+      view.setMovementMethod(LinkMovementMethod.getInstance());
+      builder.setView(view);
+      builder.setCancelable(false);
+      builder.setPositiveButton(R.string.yes, (d, w) -> openTerminal());
```

Comment:
The positive and negative button's text should be 'Ok' (`android.R.string.ok`) and 'Cancel' (`android.R.string.cancel`) instead of 'Yes' and 'No'.